src/qibolab/instruments/dummy.py

```diff
@@ -68,13 +68,38 @@ def get_values(self, options, sequence, exp_points):
                 values = np.random.rand(samples * exp_points) * 100 + 1j * np.random.rand(samples * exp_points) * 100
             elif options.acquisition_type is AcquisitionType.INTEGRATION:
                 values = np.random.rand(exp_points) * 100 + 1j * np.random.rand(exp_points) * 100
+        return values
+
+    def play(
+        self,
+        qubits: Dict[Union[str, int], Qubit],
+        sequence: PulseSequence,
+        options: ExecutionParameters,
+    ):
+        exp_points = 1 if options.averaging_mode is AveragingMode.CYCLIC else options.nshots
+        results = {}
+
+        for ro_pulse in sequence.ro_pulses:
+            values = self.get_values(options, sequence, exp_points)
             results[ro_pulse.qubit] = results[ro_pulse.serial] = options.results_type(values)
+
         return results
 
-    def play(self, qubits: Dict[Union[str, int], Qubit], sequence: PulseSequence, options: ExecutionParameters):
+    def play_sequences(
+        self,
+        qubits: Dict[Union[str, int], Qubit],
+        sequence: List[PulseSequence],
+        options: ExecutionParameters,
+    ):
         exp_points = 1 if options.averaging_mode is AveragingMode.CYCLIC else options.nshots
+        results = defaultdict(list)
+        for small_sequence in sequence:
```

Comment:
If you change the argument to `sequences` you can then do `for sequence in sequences`